### ISSUE TITLE
[Bug]: DataGridProps is any

### ISSUE THREAD

**Issue by cemcakirlar** [microsoft/fluentui#33485]
### Component

DataGrid

### Package version

9.56.6

### React version

19.0.0

### Environment

```shell
System:
    OS: macOS 15.1.1
    CPU: (12) arm64 Apple M4 Pro
    Memory: 1.22 GB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Edge: 131.0.2903.99
    Safari: 18.1.1
```

### Current Behavior

DataGrid component is typed as any. Cannot get autocomplete in VSCode for properties.

### Expected Behavior

It should be typed correctly.

### Reproduction

The issue seems trivial.

### Steps to reproduce

Use DataGrid and try to add props in a typescript project. Mine is a NextJS project

### Are you reporting an Accessibility issue?

None

### Suggested severity

High - No workaround

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by cemcakirlar** [comment]
How it looks in VSCode :)

<img width="552" alt="Image" src="https://github.com/user-attachments/assets/2ae2aae3-d394-4a49-8e08-9ab9c6b3fde2" />

**Comment by layershifter** [comment]
Duplicate, see https://github.com/microsoft/fluentui/issues/33482#issuecomment-2550856367.

**Comment by cemcakirlar** [comment]
thank you for showing me the right issue, one comment I want to add is, people may start searching type checks, typing issue, props are not visible etc. I accept that I did not made a rigorous search for the issue but I just saw that other components are also affected the same way. hence the datagrid mention. I suggest that the real issue should contain some right keywords to pin point. again thanks for the help.
